addons/hr_expense/models/hr_expense.py

```diff
@@ -845,7 +845,7 @@ def _check_employee(self):
     @api.constrains('expense_line_ids', 'company_id')
     def _check_expense_lines_company(self):
         for sheet in self:
-            if not all(expense.company_id == sheet.company_id for expense in sheet.expense_line_ids):
+            if any(expense.company_id != sheet.company_id for expense in sheet.expense_line_ids):
```

Comment:
Just curious. Comparing recordset is ok? rather ID ( expense.company_id.id )

Reply:
Yes, ``__eq__`` is implemented on Odoo recordset:
```python
def __eq__(self, other):
        """ Test whether two recordsets are equivalent (up to reordering). """
        if not isinstance(other, BaseModel):
            if other:
                filename, lineno = frame_codeinfo(currentframe(), 1)
                _logger.warning("Comparing apples and oranges: %r == %r (%s:%s)",
                                self, other, filename, lineno)
            return NotImplemented
        return self._name == other._name and set(self._ids) == set(other._ids)
```

Reply:
Yes, thats for sure in core ORM. 
But comparing IDs will have a performance edge, isnt it?
__eq__ ultimately calls self._ids, other.ids !
cc @pedrobaeza

Reply:
@JayVora-SerpentCS what do you want from me? If it's to see this, I see perfectly OK and the same performant to compare recordsets and I do it frequently (in fact, I recommend in my reviews to remove `.id` from comparisons). Anyway, @Feyensv is one of the hardcore ORM developers and he knows well what he is doing.

Reply:
thats great, just wanted an opinion! Thanks.

Reply:
The only scenario/scope where comparing records instead of ids can be dangerous/wrong is in the scope of onchange (& New records in general):
```python
NewRecord(origin=1) != Record(id=1)
NewRecord(origin=1).id != Record(id=1).id
NewRecord(origin=1)._origin.id == Record(id=1).id
NewRecord(origin=1).ids == Record(id=1).ids
```